cmd/gcr/main.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/gcr"
+	"github.com/spf13/cobra"
+)
+
+var cmds = &cobra.Command{
+	Use:   "gcr",
+	Short: "gcr is a tool for managing container images on gcr.io",
+	Run:   func(cmd *cobra.Command, _ []string) { cmd.Usage() },
+}
+
+func main() {
+	// Use google-specific implementations for listing and gc.
+	cmds.AddCommand(gcr.NewCmdGc())
+	cmds.AddCommand(gcr.NewCmdList())
+
+	// Use crane for everthing else so that this can be a drop-in replacement.
+	cmds.AddCommand(crane.NewCmdAppend())
```

Comment:
This will need to be kept in sync with `cmd/crane/main.go`, which isn't great. Can we somehow convince cobra to let us passthrough unknown commands to another tool? Or grab them and pass them through ourselves?

Reply:
Done.